android/core/src/main/java/com/stadiamaps/ferrostar/core/FerrostarCore.kt

```diff
@@ -226,6 +234,9 @@ class FerrostarCore(
   ): DefaultNavigationViewModel {
     stopNavigation()
 
+    // Start the foreground notification service
+    foregroundServiceManager.startService { stopNavigation() }
```

Comment:
I _think_ Kotlin lets you do higher-order function passing like this.

```suggestion
    foregroundServiceManager.startService(stopNavigation)
```

Reply:
Almost! 😄